src/ResolverFactory/ReflectionAttributeResolverFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Hydrator\ResolverFactory;
+
+use Yiisoft\Hydrator\DataAttributeInterface;
+use Yiisoft\Hydrator\ObjectFactory\ObjectFactoryInterface;
+use Yiisoft\Hydrator\ParameterAttributeInterface;
+
+use function is_string;
+
+final class ReflectionAttributeResolverFactory implements AttributeResolverFactoryInterface
+{
+    public function __construct(
+        private ObjectFactoryInterface $objectFactory,
```

Comment:
This is abstract dependency, user may combine it with any they like.

Reply:
<img width="1111" alt="image" src="https://github.com/yiisoft/hydrator/assets/6815714/8287a959-1493-4644-b0d5-375659f734d0">
Without the checks on the picture `newInstance()` fails with `Fatal error`. I don't want just copy-paste these check here, I prefer to use a composition with two classes. But one of classes has a public interface and I don't see any reasons to lock `ReflectionObjectFactory` with `ReflectionAttributeResolverFactory` if it may be widely-used with **any other** `AttributeResolverFactoryInterface` implementations in pair with `ReflectionAttributeResolverFactory`

Reply:
Here not need **any other** implementation. It's specific implementation, that use `ReflectionObjectFactory`.